### DISCUSSION TITLE
RaspberryMatic 3.77.6.20240720

### DISCUSSION THREAD

**Discussion by jens-maus** [OpenCCU/OpenCCU#2806] [Announcements]
This is release 3.77.6.20240720 of RaspberryMatic which is a **major** release with the following bugfixes and feature changes:

[![Downloads](https://img.shields.io/github/downloads/jens-maus/RaspberryMatic/3.77.6.20240720/total.svg?style=flat-square)](https://github.com/jens-maus/RaspberryMatic/releases/3.77.6.20240720) ![License](https://img.shields.io/github/license/jens-maus/RaspberryMatic.svg?style=flat-square) [![Donate](https://img.shields.io/badge/donate-PayPal-green.svg?style=flat-square)](https://www.paypal.com/cgi-bin/webscr?cmd=_s-xclick&hosted_button_id=RAQSDY9YNZVCL) [![GitHub sponsors](https://img.shields.io/static/v1?label=Sponsor&message=%E2%9D%A4&logo=GitHub&link=https://github.com/sponsors/jens-maus)](https://github.com/sponsors/jens-maus)

<sub>[Hier klicken für deutschsprachige 🇩🇪🇦🇹🇨🇭 Version des ChangeLogs/Diskussionsbeitrag](https://homematic-forum.de/forum/viewtopic.php?f=65&t=82946)</sub>

## 🚧 Changes:
<sub>For all changes, see the [full commit log](https://github.com/jens-maus/RaspberryMatic/compare/3.75.7.20240601...3.77.6.20240720).</sub>

#### CCU/HomeMatic service changes:
- updated [OCCU](https://github.com/eq-3/occu) version to [3.77.6-1](https://github.com/jens-maus/occu/tree/b_3_77) integrating full compatibility to the [CCU3 3.77.6 firmware](https://homematic-ip.com/sites/default/files/other_media/ccu3-changelog-3-77-6.pdf) which comes with the following changes:
   - HmIP-PSM-x: Missing unit added for certain configuration values.
   - When linking HmIP roller shutter/blind actuators with a remote control, the corresponding button had to be pressed twice to raise the shutter.
   - HmIP-BL-x - After switching from blinds to shutters, the actuator was still displayed as blinds under Status/Operation was still displayed as blinds.
   - If more than one HmIP-WGD(-PL) has been taught to the CCU3, the wrong channels may have been offered in the channel selection for links, the wrong channels may have been offered.
   - If more than one HmIP-RGBW is connected to the CCU3, incorrect channels may have been incorrect channels may have been offered for links.
   - HmIP-ESI - After a meter change, the history values of the IEC sensor were not adjusted to the new meter.
   - HmIP-DRG-DALI - An error in the Easymode profile “Brightness (V) - on/brighter” meant that dimming was only possible in steps rather than continuously when a button was pressed for a long time.
   - Missing JSON-API function reportValueUsage added.
   - For wall thermostats with two-point control, the resolution of the hysteresis value in “Hygrostat” operating mode has been changed from 2% to 1%.
   - HmIP-BWTH/FAL/FALMOT - The adjustable values for the switch-on/switch-off delay and the interval for the descaling run have been changed for the Boler and pump heat requirement channels.
- ⚠️ **BREAKING CHANGE** ⚠️: the `intelnuc` platform had been renamed to `generic-x86_64` and changed from old Legacy BIOS to (U)EFI BIOS mode for more modern and generic x86 hardware compatibility not just limited to IntelNUC based systems. Existing users who currently use and updated their `intelnuc` platform images would have to perform a fresh image-based reinstall using the `generic-x86_64` images after changing their systems to UEFI boot mode and then simply import their latest sbk-Backup to restore their CCU setup. (#2759).
- update Mediola NEOserver to latest 2.16.1.
- when running as a HA addon the daily cronBackup will now place CCU `*.sbk` type backups to `/backup/raspberrymatic` to utilize the standard HA backup path.
- rework data path to `/usr/local` mapping in a HA addon use case by using a newer possibility by HA to define the target path directly in the addon `config.yaml`. This allows to remove the `S00DataMount` init script to bind mount `/data` to `/usr/local` as a workaround (#2800).

#### WebUI changes:
- add another WebUI fix to the `0039-WebUI-Fix-Style-Glitches` patch set which will make sure the rooms and functions popup will come up properly centered rather than using a fixed height.
- add new `0185-WebUI-Fix-CPPopupCentering` WebUI patch to fix broken control panel popup centering for the time and software control panel (#2775).

#### Operating system changes:
- update tailscale VPN to 1.70.0.
- update upstream kernel to 6.6.41 LTS.
- update buildroot openssh package to 9.8p1 to fix CVE-2024-6387.
- add rpi kernel upstream patch fixing BLK exceptions with `dwc_otg` when compiled with GCC >= 13 (#2780).
- move `/dev_host` setup from `S06InitSystem` to `S01InitHost` and added rtc device node symlinking for the LXC/OCI platform case so that upon startup an already available RTC clock can be correctly identified.
- change `RPI-RF-MOD` GPIO pin retrieval for the LXC/OCI platform use case which should fix issue with kernel changing the pin id (#2763).
- ensure that the symlink for `/etc/hosts` is not removed for the LXC platform so that in case rootfs is read-only a valid `/etc/hosts` file can be generated in the `/var` tmpfs upon bootup. This should resolve issues with incorrectly working DNS requests for "localhost" in case of an LXC container setup.
- add `lxc.signal.stop = SIGTERM` definition to global lxc container config to ensure that the correct signal command is used when the host is shutdown and stops all LXC containers (#2762).
- remove leading "-" from abi.cp15_barrier in sysctl.conf to make sure this sysctl statement is used. This reverts the changes proposed by https://github.com/home-assistant/operating-system/pull/3285.
- update wiringpi-rpi to latest 3.6.
- update rpi-eeprom to latest pieeprom-2024-06-05.bin.
- update qemu-guest-agent to latest 9.0.2.
- update buildroot to latest 2024.05.1.
- update rpi-firmware to latest version (1.20240529).
- update rpi-kernel to latest stable_20240529 (6.6.31).

## 👪 Contributors (alphabetically):
- @jens-maus

## 📝 Support:
For support on installation and help please visit the following web pages:

[RaspberryMatic - Documentation :us:](https://github.com/jens-maus/RaspberryMatic/wiki/en.Home)
[RaspberryMatic - Discussions :us:](https://github.com/jens-maus/RaspberryMatic/discussions)
[RaspberryMatic - Dokumentation :de:](https://github.com/jens-maus/RaspberryMatic/wiki)
[RaspberryMatic - Forum :de:](https://homematic-forum.de/forum/viewforum.php?f=65)

## 📦 Download:
The following installation archives can be downloaded for different hardware platforms. To verify their integrity a `sha256` checksum is provided as well. You can either upload these files using the WebUI-based update mechanism or unarchive them to e.g. flash the included `*.img` files on a fresh installation media (e.g. microSD card):

- RaspberryPi5 Model B – ([installation](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-RaspberryPi)):
<sub>📦 [RaspberryMatic-3.77.6.20240720-rpi5.zip](https://github.com/jens-maus/RaspberryMatic/releases/download/3.77.6.20240720/RaspberryMatic-3.77.6.20240720-rpi5.zip)<br/>SHA256: 8366e6cf0b18ec1723374f6e14fa4aac3d30541c94e7973075133698cd8d578a</sub>

- RaspberryPi4 Model B, RaspberryPi Compute Module 4, RaspberryPi 400 – ([installation](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-RaspberryPi)):
<sub>📦 [RaspberryMatic-3.77.6.20240720-rpi4.zip](https://github.com/jens-maus/RaspberryMatic/releases/download/3.77.6.20240720/RaspberryMatic-3.77.6.20240720-rpi4.zip)<br/>SHA256: 948911b413ecba56e6499251abce5cf3f4315c358bf973f404ccfb33f520f769</sub>

- [CCU3](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-CCU3), [ELV-Charly](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-ELV-Charly), RaspberryPi3 Model B+, RaspberryPi3 Model B, RaspberryPi3 Model A+, RaspberryPi Compute Module 3, RaspberryPi Compute Module 3 lite, RaspberryPi Zero 2 W – ([installation](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-RaspberryPi)):
<sub>📦 [RaspberryMatic-3.77.6.20240720-rpi3.zip](https://github.com/jens-maus/RaspberryMatic/releases/download/3.77.6.20240720/RaspberryMatic-3.77.6.20240720-rpi3.zip)<br/>SHA256: da76f9441a6deb204fb59059978fb59f6f59501194f10c303369aece23e17392</sub>
<sub>📦 [RaspberryMatic-3.77.6.20240720-ccu3.tgz](https://github.com/jens-maus/RaspberryMatic/releases/download/3.77.6.20240720/RaspberryMatic-3.77.6.20240720-ccu3.tgz) (only for initial CCU3 Firmware -> RaspberryMatic Upgrade)<br/> SHA256: 85ea08b858ed6c06dbca943cfb8999e66816d6e43fef5cb598551363829538a3</sub>

- RaspberryPi2 Model B – ([installation](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-RaspberryPi)):
<sub>📦 [RaspberryMatic-3.77.6.20240720-rpi2.zip](https://github.com/jens-maus/RaspberryMatic/releases/download/3.77.6.20240720/RaspberryMatic-3.77.6.20240720-rpi2.zip)<br/>SHA256: 5c3ef53145e4507d2ec5111d563b8e5f229db4cd70c7da07ddd8b15b40fba181</sub>

- RaspberryPi Zero W, RaspberryPi Zero, RaspberryPi Compute Module 1, RaspberryPi1 (A+/B+) – ([installation](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-RaspberryPi)):
<sub>📦 [RaspberryMatic-3.77.6.20240720-rpi0.zip](https://github.com/jens-maus/RaspberryMatic/releases/download/3.77.6.20240720/RaspberryMatic-3.77.6.20240720-rpi0.zip)<br/>SHA256: 05b56b047cf453ae2103a85baad5bbd31b8396a84b039adf98b5a808f4177b02</sub>

- TinkerBoard S, TinkerBoard – ([installation](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-Tinkerboard)):
<sub>📦 [RaspberryMatic-3.77.6.20240720-tinkerboard.zip](https://github.com/jens-maus/RaspberryMatic/releases/download/3.77.6.20240720/RaspberryMatic-3.77.6.20240720-tinkerboard.zip)<br/>SHA256: b5ddba00abbd2d3d1c42bfed20e47977aad105d26f20cf0b54dd81e163ffd22d</sub>

- ODROID-N2/N2+/C4/C2 – ([installation](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-Odroid)):
<sub>📦 [RaspberryMatic-3.77.6.20240720-odroid-n2.zip](https://github.com/jens-maus/RaspberryMatic/releases/download/3.77.6.20240720/RaspberryMatic-3.77.6.20240720-odroid-n2.zip)<br/>SHA256: 2291ab46f835386fe8e58a8ed4bc9d27814f33c24846304d61998afe2826895c</sub>
<sub>📦 [RaspberryMatic-3.77.6.20240720-odroid-c4.zip](https://github.com/jens-maus/RaspberryMatic/releases/download/3.77.6.20240720/RaspberryMatic-3.77.6.20240720-odroid-c4.zip)<br/>SHA256: 3dbad5b6984f2245cb167ad49c8a14124fab5f14a0bf3ffe84876ba4dd6deb5e</sub>
<sub>📦 [RaspberryMatic-3.77.6.20240720-odroid-c2.zip](https://github.com/jens-maus/RaspberryMatic/releases/download/3.77.6.20240720/RaspberryMatic-3.77.6.20240720-odroid-c2.zip)<br/>SHA256: 7c394def0cee99343722a6571ac93abf82786d4a33aba3ef020151f309255fac</sub>

- Generic-x86_64 – ([installation](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-IntelNUC)):
<sub>📦 [RaspberryMatic-3.77.6.20240720-generic-x86_64.zip](https://github.com/jens-maus/RaspberryMatic/releases/download/3.77.6.20240720/RaspberryMatic-3.77.6.20240720-generic-x86_64.zip)<br/>SHA256: 4f1a21aed961373143ceaf740caeba25b0e8a2027f479cc3bf368d300d72b370</sub>

- Open Virtual Appliance (OVA) – ([ProxmoxVE](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-Proxmox-VE), [VirtualBox](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-VirtualBox), [ESXi](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-vmWare-ESXi), [Synology](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-Synology-VMM), [QNAP](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-QNAP-VirtualizationStation), [Workstation Player](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-vmWare-Workstation-Player), [QEmu](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-QEmu), [UNRAID](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-UNRAID), [HyperV](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-HyperV)):
<sub>📦 [RaspberryMatic-3.77.6.20240720-ova.zip](https://github.com/jens-maus/RaspberryMatic/releases/download/3.77.6.20240720/RaspberryMatic-3.77.6.20240720-ova.zip)<br/>SHA256: efac8d64ff83f91da0a05010b7f56c0da23dd1cc364398d62c0d2897c8ba9963</sub>
<sub>📦 [RaspberryMatic-3.77.6.20240720.ova](https://github.com/jens-maus/RaspberryMatic/releases/download/3.77.6.20240720/RaspberryMatic-3.77.6.20240720.ova) (only for initial OVA installation)<br/>SHA256: 27a1a94070bbf1027e17811020f36fb89d616a90c87f4a2818ab1a20a10b3d85</sub>

- Docker / Open Container Initiative (OCI) – virtual appliance ([installation](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-Docker-OCI)):
<sub>📦 [RaspberryMatic-3.77.6.20240720-oci_amd64.tgz](https://github.com/jens-maus/RaspberryMatic/releases/download/3.77.6.20240720/RaspberryMatic-3.77.6.20240720-oci_amd64.tgz) (amd64/x86_64)<br/>SHA256: 95d65915147305616456352c6cb00feb53e0ceb42c13ae6b38b78512b6f747bc</sub>
<sub>📦 [RaspberryMatic-3.77.6.20240720-oci_arm64.tgz](https://github.com/jens-maus/RaspberryMatic/releases/download/3.77.6.20240720/RaspberryMatic-3.77.6.20240720-oci_arm64.tgz) (arm64/aarch64)<br/>SHA256: e468c6010dfe100af25c46a256af1456fc43ca0e09733c6ff7dfcc987ecd6d1c</sub>
<sub>📦 [RaspberryMatic-3.77.6.20240720-oci_arm.tgz](https://github.com/jens-maus/RaspberryMatic/releases/download/3.77.6.20240720/RaspberryMatic-3.77.6.20240720-oci_arm.tgz) (arm/armhf)<br/>SHA256: 9440c3debc12410f9181a1551ac659685f1dad9c01fec1ee307d85728618c9fa</sub>

- LXC Container – virtual appliance ([installation](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-LXC)):
<sub>📦 [RaspberryMatic-3.77.6.20240720-lxc_amd64.tar.xz](https://github.com/jens-maus/RaspberryMatic/releases/download/3.77.6.20240720/RaspberryMatic-3.77.6.20240720-lxc_amd64.tar.xz) (amd64/x86_64)<br/>SHA256: 27fe13330ccea9a294db524756a1412474affe5cd0988e68cd04f46465ff88cf</sub>
<sub>📦 [RaspberryMatic-3.77.6.20240720-lxc_arm64.tar.xz](https://github.com/jens-maus/RaspberryMatic/releases/download/3.77.6.20240720/RaspberryMatic-3.77.6.20240720-lxc_arm64.tar.xz) (arm64/aarch64)<br/>SHA256: 94e372502ae66d997a2b058e7b31f79002eb49d529b1ee614daeb5e66080e511</sub>
<sub>📦 [RaspberryMatic-3.77.6.20240720-lxc_arm.tar.xz](https://github.com/jens-maus/RaspberryMatic/releases/download/3.77.6.20240720/RaspberryMatic-3.77.6.20240720-lxc_arm.tar.xz) (arm/armhf)<br/>SHA256: b70d360a9d69eeafeb8e9cd86b373e77ddff86a49c5136690e366d72be983e73</sub>

- Generic-aarch64 Appliance – ([ProxmoxVE](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-Proxmox-VE)):
<sub>📦 [RaspberryMatic-3.77.6.20240720-generic-aarch64.zip](https://github.com/jens-maus/RaspberryMatic/releases/download/3.77.6.20240720/RaspberryMatic-3.77.6.20240720-generic-aarch64.zip)<br/>SHA256: 64422ef26e787f580060d3aad48cadba40a8b450751f6493294c20ed840493ee</sub>

- Kubernetes / K8s – virtual appliance:
<sub>see [install documentation](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-Kubernetes)</sub>

- Home Assistant Add-on – virtual appliance:
<sub>see [install documentation](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-HomeAssistant)</sub>

<hr /><em>This discussion was created from the release <a href='https://github.com/jens-maus/RaspberryMatic/releases/tag/3.77.6.20240720'>RaspberryMatic 3.77.6.20240720</a>.</em>

**Comment by solomal1** [comment]
Hi, 

I've gone carefully through the update process for RaspberryMatic CCU Installed version 3.75.7.20240601 to Latest version 3.77.6.20240720 twice now with the same result **- all my devices are lost**

The update is being performed against:
Home Assistant Core 2024.8.1 / Supervisor 2024.08.0 / Operating System 13.0 /Frontend 20240809.0

I've also tried taking a backup .sbk file from the Security settings and restoring that after the update and still no devices. (Also checked after full reboot)

Surely there must be a step I am missing - any advice greatly appreciated!

Alan
